perfkitbenchmarker/linux_benchmarks/specsfs2014_benchmark.py

```diff
@@ -0,0 +1,264 @@
+  config_path = posixpath.join(SPEC_DIR, 'sfs_rc')
+
+  if FLAGS.specsfs2014_config:
+    prime_client.PushFile(
+        data.ResourcePath(FLAGS.specsfs2014_config), config_path)
+  else:
+    stdout, _ = prime_client.RemoteCommand('pwd')
+    exec_path = posixpath.join(stdout.strip(), SPEC_DIR, 'binaries',
+                               'linux', 'x64', 'netmist')
+    configuration_overrides = {
+        'USER': prime_client.user_name,
+        'EXEC_PATH': exec_path.replace('/', '\/'),
+        'CLIENT_MOUNTPOINTS': 'mountpoints.txt',
+        'BENCHMARK': FLAGS.specsfs2014_benchmark,
+        'LOAD': FLAGS.specsfs2014_load,
+        'NUM_RUNS': FLAGS.specsfs2014_num_runs,
+        'INCR_LOAD': FLAGS.specsfs2014_incr_load
+    }
+
+    sed_expressions = ' '.join(
+        ['-e "s/{0}=.*/{0}={1}/"'.format(k, v)
+         for k, v in configuration_overrides.iteritems()])
+    sed_cmd = 'sudo sed -i {0} {1}'.format(sed_expressions, config_path)
+    prime_client.RemoteCommand(sed_cmd)
+
+  with vm_util.NamedTemporaryFile() as tf:
+    for client in clients:
+      tf.write('%s %s\n' % (client.internal_ip, MOUNT_POINT))
+    tf.close()
+    prime_client.PushFile(tf.name, posixpath.join(SPEC_DIR, 'mountpoints.txt'))
+
+
+def Prepare(benchmark_spec):
+  """Install SPEC SFS 2014.
+
+  Args:
+    benchmark_spec: The benchmark specification. Contains all data that is
+        required to run the benchmark.
+  """
+  gluster_servers = benchmark_spec.vm_groups['gluster_servers']
+  clients = benchmark_spec.vm_groups['clients']
+  prime_client = clients[0]
+
+  vm_util.RunThreaded(_PrepareServer, gluster_servers)
+  gluster.ConfigureServers(gluster_servers, VOLUME_NAME)
+
+  server = gluster_servers[0] if gluster_servers else None
+  args = [((client, server), {}) for client in clients]
+  vm_util.RunThreaded(_PrepareClient, args)
+
+  _ConfigureSpec(prime_client, clients)
+
+  if not prime_client.is_static:
+    # If the prime client is a static VM, we don't want to clobber their
+    # ssh config or private key and they should have already set up
+    # passwordless ssh between instances.
+    prime_client.AuthenticateVm()
+    with vm_util.NamedTemporaryFile() as tf:
+      tf.write('Host *\n')
+      tf.write('  StrictHostKeyChecking no\n')
+      tf.close()
+      prime_client.PushFile(tf.name, '~/.ssh/config')
+
+
+def _ParseSpecSfsOutput(output, extra_metadata=None):
+  """Returns samples generated from the output of SPEC SFS 2014."""
+  root = xml.etree.ElementTree.fromstring(output)
+  samples = []
+
+  for run in root.findall('run'):
+    metadata = {
+        'benchmark': run.find('benchmark').attrib['name'],
+        'business_metric': run.find('business_metric').text
+    }
+    if extra_metadata:
+      metadata.update(extra_metadata)
+
+    for key in METADATA_KEYS:
+      element = run.find('metric[@name="%s"]' % key)
+      units = element.attrib.get('units')
+      label = (key + ' (%s)' % units) if units else key
```

Comment:
I think I was trying to make it fit in one line... but it looks like it fits even with one format so I'm not sure what I was thinking...